kotlinx-coroutines-core/common/test/flow/operators/DebounceTest.kt

```diff
@@ -223,4 +221,68 @@ class DebounceTest : TestBase() {
         assertEquals(listOf("A", "D", "E"), result)
         finish(5)
     }
+
+    @Test
+    fun testDebounceSelectorBasic() = withVirtualTime {
+        expect(1)
+        val flow = flow {
+            expect(3)
+            emit("A")
+            delay(1500)
+            emit("B")
+            delay(500)
+            emit("C")
+            delay(1)
+            emit("D")
+            delay(800)
+            emit("E")
+            expect(4)
+        }
+
+        expect(2)
+        val result = flow.debounce {
+            if (it == "C") {
+                0
+            } else {
+                1000
+            }
+        }.toList()
+
+        assertEquals(listOf("A", "C", "E"), result)
+        finish(5)
+    }
+
+    @Test
+    fun testZeroDebounceTime() = withVirtualTime {
+        expect(1)
+        val flow = flow {
+            expect(2)
+            emit("A")
+            emit("B")
+            emit("C")
+            expect(3)
+        }
+
+        val result = flow.debounce(0).toList()
+
+        assertEquals(listOf("A", "B", "C"), result)
+        finish(4)
+    }
+
+    @Test
+    fun testZeroDebounceTimeSelector() = withVirtualTime {
+        expect(1)
+        val flow = flow {
+            expect(2)
+            emit("A")
+            delay(1)
```

Comment:
When I emit two values immediately one after another, the first one never reaches the `select<Unit> {` block. I see the same behavior prior to my changes though. 

It looks like it's getting conflated, but I don't understand where exactly. I tried modifying `val values = produce<Any?>(capacity = Channel.CONFLATED) {` to use different capacity just to troubleshoot, but no luck.

Any suggestions are appreciated.

Reply:
Changing from `produce<Any?>(capacity = Channel.CONFLATED)` to `produce<Any?>(capacity = 0)` should remove the conflation. The reason that `CONFLATED` is used here, though, is because the `debounce` is effectively a "stronger conflation" operator that conflates even more values.

Reply:
It still fails the `testZeroDebounceTimeSelector` test with `kotlin.AssertionError: Expected <[A, B]>, actual <[B]>` if I remove that `delay(1)` from my test code.

Reply:
I don't understand why `testZeroDebounceTimeSelector` fails without that `delay(1)`. Any theories?